src/main/utils/settings.ts

```diff
@@ -380,12 +381,15 @@ export class Settings<T extends object> extends PropertyChangeEmitter<T> {
         Object.prototype.hasOwnProperty.call(newSettings, key) &&
         !lodash.isEqual(newSettings[key], oldSettings[key])
       ) {
-        this.emit(key, this.settings[key], oldSettings[key]);
+        this.emit(key, newSettings[key], oldSettings[key]);
         anyChanged = true;
       }
     }
 
     if (anyChanged) {
+      if (!lodash.isEqual(oldSettings['locale'], newSettings['locale'])) {
+        i18next.changeLanguage(newSettings['locale']);
+      }
```

Comment:
I think this can be removed, right? Changing languages on-the-fly does not work anyways.

```suggestion
```